File: typescript/src/renderer.ts
import { Shape } from './shapes/shape.js'

function newRenderer (shape: Shape): {draw: () => void} {
  return {
    draw () {
      const shapeType: string = shape.getType()
      const area: number = shape.computeArea()
      console.log(`${shapeType} drawn\n` + `Its area is ${area}`)
    }
  }
}

export { newRenderer }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.*;

public class Renderer {
    private Shape shape;
    
    Renderer(Shape shape) {
        this.shape = shape;
    }

    void draw() {
        String shapeType = shape.getType();
        double area = shape.getArea();
        System.out.println(shapeType + " printed\n" + "Its area is " + area);
    }
}
